scene/main/scene_tree.cpp

```diff
@@ -2070,8 +2069,10 @@ SceneTree::SceneTree() {
 	last_screen_size = Size2(OS::get_singleton()->get_window_size().width, OS::get_singleton()->get_window_size().height);
 	_update_root_rect();
 
-	if (ScriptDebugger::get_singleton()) {
-		ScriptDebugger::get_singleton()->set_request_scene_tree_message_func(_debugger_request_tree, this);
+	if (ScriptDebugger::get_singleton() && ScriptDebugger::get_singleton()->is_remote()) {
```

Comment:
There probably is no need for SceneTree to know about the debugger, I think Debugger should take care of everything.

Reply:
This is only for Debugger to get `SceneTree` pointer to do all its business. Where do you propose I get this pointer otherwise? On a similar note, `SceneTree` still needs access to `ScriptDebugger` for quitting game when F8 is hit. Do you think this could be offloaded to `ScriptDebuggerRemote` so we can get rid of all `SceneTree` -> `ScriptDebugger` coupling?

Reply:
Either way, I'd make the shortcut to quit the game programmable in editor shortcuts, having it hardcoded can be problematic for the users.

Reply:
I would just pass it in main.cpp after a SceneTree is created, in the
```
if (main_loop->is_class("SceneTree")) {
```
conditional, and if its not the editor.